resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/v2/KubernetesSparkDependencyDownloadInitContainer.scala

```diff
@@ -61,55 +64,153 @@ private class DownloadTarGzCallback(downloadDir: File) extends WaitableCallback[
-    s" at ${downloadFilesSecretLocation.getAbsolutePath} does not exist or is not a file.")
-
-  private val jarsDownloadDir = new File(sparkConf.get(DRIVER_LOCAL_JARS_DOWNLOAD_LOCATION))
-  require(jarsDownloadDir.isDirectory, "Application jars download directory provided at" +
-    s" ${jarsDownloadDir.getAbsolutePath} does not exist or is not a directory.")
-
-  private val filesDownloadDir = new File(sparkConf.get(DRIVER_LOCAL_FILES_DOWNLOAD_LOCATION))
-  require(filesDownloadDir.isDirectory, "Application files download directory provided at" +
-    s" ${filesDownloadDir.getAbsolutePath} does not exist or is not a directory.")
+
+  private val stagingServerJarsDownloadDir = new File(
+    sparkConf.get(DRIVER_SUBMITTED_JARS_DOWNLOAD_LOCATION))
+  private val stagingServerFilesDownloadDir = new File(
+    sparkConf.get(DRIVER_SUBMITTED_FILES_DOWNLOAD_LOCATION))
+
+  private val remoteJars = sparkConf.get(INIT_CONTAINER_REMOTE_JARS)
+  private val remoteFiles = sparkConf.get(INIT_CONTAINER_REMOTE_FILES)
+  private val remoteJarsDownloadDir = new File(
+    sparkConf.get(DRIVER_REMOTE_JARS_DOWNLOAD_LOCATION))
+  private val remoteFilesDownloadDir = new File(
+    sparkConf.get(DRIVER_REMOTE_FILES_DOWNLOAD_LOCATION))
+
   private val downloadTimeoutMinutes = sparkConf.get(DRIVER_MOUNT_DEPENDENCIES_INIT_TIMEOUT)
 
   def run(): Unit = {
-    val securityManager = new SparkSecurityManager(sparkConf)
-    val sslOptions = securityManager.getSSLOptions("kubernetes.resourceStagingServer")
-    val service = retrofitClientFactory.createRetrofitClient(
-      resourceStagingServerUri, classOf[ResourceStagingServiceRetrofit], sslOptions)
-    val jarsSecret = Files.toString(downloadJarsSecretLocation, Charsets.UTF_8)
-    val filesSecret = Files.toString(downloadFilesSecretLocation, Charsets.UTF_8)
-    val downloadJarsCallback = new DownloadTarGzCallback(jarsDownloadDir)
-    val downloadFilesCallback = new DownloadTarGzCallback(filesDownloadDir)
-    service.downloadResources(downloadJarsResourceIdentifier, jarsSecret)
-      .enqueue(downloadJarsCallback)
-    service.downloadResources(downloadFilesResourceIdentifier, filesSecret)
-      .enqueue(downloadFilesCallback)
-    logInfo("Waiting to download jars...")
-    downloadJarsCallback.waitForCompletion(downloadTimeoutMinutes, TimeUnit.MINUTES)
-    logInfo(s"Jars downloaded to ${jarsDownloadDir.getAbsolutePath}")
-    logInfo("Waiting to download files...")
-    downloadFilesCallback.waitForCompletion(downloadTimeoutMinutes, TimeUnit.MINUTES)
-    logInfo(s"Files downloaded to ${filesDownloadDir.getAbsolutePath}")
+    val resourceStagingServerJarsDownload = Future[Unit] {
+      downloadResourcesFromStagingServer(
+        maybeDownloadJarsResourceIdentifier,
+        downloadJarsSecretLocation,
+        stagingServerJarsDownloadDir,
+        "Starting to download jars from resource staging server...",
+        "Finished downloading jars from resource staging server.",
```

Comment:
change this two starting/finished message params to just "jars" or "files" and do the logging in the method.  This also lets us drop the last two failure message parameters too right?